arduino/src/arduino.cpp

```diff
@@ -0,0 +1,214 @@
+/**
+ * @file arduino.cpp
+ * @author yarin
+ * @brief  the arduino<-> pi protocol arduino object
+ * @version 0.1
+ * @date 2020-11-21
+ * @copyright Copyright (c) 2020
+ * 
+ */
+
+#include <ctime>
+#include "arduino.h"
+#define CHECK_BUFFER_SIZE 64
+#define FILE_NAME "../dataFromArduino/sensor"
+#define FILE_NAME_END ".log"
+
+Arduino::~Arduino()
+{
+    this->serial.closeDevice();
+}
+
+Arduino::Arduino(const std::string &portName, unsigned int buadRate, unsigned int timeout) : portName(portName), buadRate(buadRate), timeout(timeout)
+{
+}
+
+const std::vector<std::string> &Arduino::getFiles()
+{
+    return this->files;
+}
+
+int Arduino::getTimout()
+{
+    return this->timeout;
+}
+
+void Arduino::setTimeout(int timeout)
+{
+    this->timeout = timeout;
+}
+
+Arduino::ERROR_ARDUINO Arduino::checkConnection()
+{
+    if (!this->isOpen)
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    char helper = serial.openDevice(portName.c_str(), buadRate);
+    if (helper != 1)
+    {
+        serial.closeDevice();
+    }
+
+    serial.closeDevice();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+void Arduino::checkConnectionToConsole()
+{
+    if (checkConnection() != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        std::cout << "there was a problem with the connection" << std::endl;
+    }
+
+    std::cout << "the connection is successful!-" << std::endl;
+}
+
+void Arduino::addFile(const std::string &file)
+{
+    this->files.push_back(file);
+}
+
+int Arduino::checkSum(const char *message)
+{
+    int sum = 0;
+    const char *p = &message[0];
+    while (*p != NULL)
+    {
+        sum += *message;
+        ++p;
+    }
+
+    return sum;
+}
+
+Arduino::ERROR_ARDUINO Arduino::openSerial()
+{
+    int err = serial.openDevice(this->portName.c_str(), this->buadRate);
+    if (err != Arduino::ERROR_ARDUINO::SUCCSESS)
```

Comment:
```suggestion
    if (err != 1)
```
because it's the liberty return codes

Reply:
fixed